care/facility/models/facility.py

```diff
@@ -69,47 +70,19 @@ class Facility(FacilityBaseModel):
     corona_testing = models.BooleanField(default=False)
     created_by = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, blank=True)
 
+    users = models.ManyToManyField(
+        User, through="FacilityUser", related_name="facilities", through_fields=("facility", "user")
+    )
+
     class Meta:
         verbose_name_plural = "Facilities"
 
     def __str__(self):
         return f"{self.name}"
 
-    @staticmethod
-    def has_bulk_upsert_permission(request):
-        return request.user.is_superuser
-
-    @staticmethod
-    def has_read_permission(request):
-        return True
-
-    def has_object_read_permission(self, request):
-        return (
-            request.user.is_superuser
-            or request.user == self.created_by
-            or (
-                request.user.user_type >= User.TYPE_VALUE_MAP["DistrictLabAdmin"]
-                and request.user.district == self.district
-            )
-        )
-
-    @staticmethod
-    def has_write_permission(request):
-        return True
-
-    def has_object_write_permission(self, request):
+    def has_object_destroy_permission(self, request):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5345345)